### PR TITLE
[gcs] Fix can not resue actor name in the same namespace after be…

### PR DESCRIPTION
**Pull request by newmanwang**
## Why are these changes needed?

   Before this PR, `GcsActorManager::CreateActor()` would replace actor's namespace by
actors's owner job's namespace, even if actor is created by user with a user specified
namespace. But in named_actors_, actor is set to use user specified namespace by
`GcsActorManager::RegisterActor` before CreateActor() is called, So that
`GcsActorManager::DestroyActor` failed to find actor from named_actors_ by owner job's
namespace to remove, hence reuse actor name in same namespace failed for same name actor
 not removed by `GcsActorManager::DestroyActor` in named_actors_.

## Related issue number
https://github.com/ray-project/ray/issues/20611

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Test in my 10 nodes gpu cluster 
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Lint failure!

**Comment by newmanwang** [comment]
Sorry,  working on it

**Comment by rkooo567** [comment]
Can you also add a cpp unit test to gcs_actor_manager.cc?

**Comment by newmanwang** [comment]
sure

**Comment by rkooo567** [comment]
Current status: Waiting for lint fix and cpp unit tests

File: src/ray/gcs/gcs_server/gcs_actor_manager.cc [hunk 1]
```diff
@@ -491,8 +491,12 @@ Status GcsActorManager::CreateActor(const ray::rpc::CreateActorRequest &request,
 
   // Remove the actor from the unresolved actor map.
   const auto job_id = JobID::FromBinary(request.task_spec().job_id());
+  const auto &actor_namespace = iter->second->GetRayNamespace();
   auto actor =
-      std::make_shared<GcsActor>(request.task_spec(), get_ray_namespace_(job_id));
+      std::make_shared<GcsActor>(
```
Comment (wuisawesome) [inline src/ray/gcs/gcs_server/gcs_actor_manager.cc:496]: Is it possible to add a c++ unit test for this? (`gcs_actor_manager_test.cc`)

**Review by wuisawesome — CHANGES_REQUESTED** [review]

**Comment by rkooo567** [comment]
Any progress on this?

**Comment by newmanwang** [comment]
> Any progress on this?

Sorry for the delay, I have been busy as hell recently. Writing cpp test now, I am hoping to submit changes tomorrow.

**Comment by newmanwang** [comment]
Could you please take a look at this clean pr https://github.com/ray-project/ray/pull/21053 and drop this one. @rkooo567

**Pull request closed without merging**